### PR TITLE
Added Profile Tests and better defined tests

### PR DESCRIPTION
**Pull request by NathanDVT**
- [X] Added Profile Unit test
- [X] Added Profile UI test
- [X] Added Fastlane skeleton

### CODE REVIEW

File: NLibrary/Classes/ProfileMVVM/Mocks/ProfileMockRepo.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public func getUserProfile(completion: @escaping(Result<ProfileModel, APIFailureResult>) -> Void) {
+        switch closurePathway {
+        case .failedRequest(_):
+            completion(.failure(.failedRequest(message: "failed getUserProfile")))
+        case .succesfullRequest:
+            let dictionary: NSDictionary = [
+                "gender": Gender.male.stringValue,
+                "name": "name",
+                "email": "email",
+                "Playlists": ["1":"x", "2": "x"]
```
Comment (NathanDVT) [inline NLibrary/Classes/ProfileMVVM/Mocks/ProfileMockRepo.swift:26]: Codacy Issue found: [Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5389739)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5389739&bid=17669980)

File: NLibrary/Classes/ProfileMVVM/Mocks/ProfileMockRepo.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            ]
+            completion(.success(ProfileModel(dictionary: dictionary)))
+        }
+    }
+
+    public func logoutRequest(completion: @escaping(APIRequestResult) -> Void) {
+        switch closurePathway {
+        case .failedRequest(_):
```
Comment (NathanDVT) [inline NLibrary/Classes/ProfileMVVM/Mocks/ProfileMockRepo.swift:34]: Codacy Issue found: [Arguments can be omitted when matching enums with associated types if they are not used.](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5389739)

**Comment by NathanDVT** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5389739&bid=17669980)

File: NLibrary/Classes/ProfileMVVM/Mocks/ProfileMockRepo.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import AVKit
+
+public class ProfileMockRepo: ProfileRepoProtocol {
+    var closurePathway: APIRequestResult = .succesfullRequest
+    public required init () {
+    }
+
+    public func getUserProfile(completion: @escaping(Result<ProfileModel, APIFailureResult>) -> Void) {
+        switch closurePathway {
+        case .failedRequest(_):
```
Comment (NathanDVT) [inline NLibrary/Classes/ProfileMVVM/Mocks/ProfileMockRepo.swift:19]: Codacy Issue found: [Arguments can be omitted when matching enums with associated types if they are not used.](https://app.codacy.com/manual/NathanDVT/NLibrary/pullRequest?prid=5389739)

**Review by JancoDVTDev — APPROVED** [review]

**Pull request merged**